File: src/main/java/view/InputView.java
package view;

import static message.GameMessage.CAR_NAMES_MESSAGE;
import static message.GameMessage.TRY_COUNT_MESSAGE;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

public class InputView {

    private static final BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
    public static final String CAR_NAME_DELIMITER = ",";

    private InputView() {

    }

    public static List<String> inputCarNames() throws IOException {
        System.out.println(CAR_NAMES_MESSAGE.getValue());
        return splitCarNames();
    }

    public static String inputTryCount() throws IOException {
        System.out.println(TRY_COUNT_MESSAGE.getValue());
        return bufferedReader.readLine();
    }

    private static List<String> splitCarNames() throws IOException {
        String line = bufferedReader.readLine();
        String[] carNames = line.split(CAR_NAME_DELIMITER);
        return Arrays.stream(carNames)
                .map(String::valueOf)
                .collect(Collectors.toList());
    }
}

File: src/main/java/view/InputView.kt
package view

import message.GameMessage
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader
import java.lang.String.valueOf
import java.util.*
import java.util.stream.Collectors

object InputView {

    private val bufferedReader = BufferedReader(InputStreamReader(System.`in`))
    private const val CAR_NAME_DELIMITER = ","

    @JvmStatic
    @Throws(IOException::class)
    fun inputCarNames(): List<String> {
        println(GameMessage.CAR_NAMES_MESSAGE.value)
        return splitCarNames()
    }

    @JvmStatic
    @Throws(IOException::class)
    fun inputTryCount(): String {
        println(GameMessage.TRY_COUNT_MESSAGE.value)
        return bufferedReader.readLine()
    }

    @Throws(IOException::class)
    private fun splitCarNames(): List<String> {
        val line = bufferedReader.readLine()
        return line.split(CAR_NAME_DELIMITER)
    }
}
